monai/utils/arrayutils.py

```diff
@@ -219,10 +219,61 @@ def iter_patch_slices(dims, patch_size, start_pos=()):
         yield tuple(slice(s, s + p) for s, p in zip(position[::-1], patch_size))
 
 
+def iter_dense_patch_slices(image_size, patch_size, scan_interval):
```

Comment:
This function isn't a generator, it returns a list. The name and docstring should be changed to match, though I would suggest we modify iter_patch_slices to fill this role instead. I was planning to look into that shortly.